lib/usd/ui/layerEditor/mayaSessionState.cpp

```diff
@@ -228,7 +220,13 @@ void MayaSessionState::nodeRenamedCBOnIdle(const MObject& obj)
 
         StageEntry entry;
         if (getStageEntry(&entry, shapePath)) {
-            Q_EMIT stageRenamedSignal(entry._displayName, entry._stage);
+            // Need to update the current Entry also
+            if (_currentStageEntry._displayName == oldName
```

Comment:
There is no garantee that the _displayName is the name we're going to get from Maya from the notifications. 
Did you test this?  What about when a proxy is in a Namespace I don't think you should be doing anything with 'old name', we already have the MObject pointer from which we can get the UsdStage, and that's how we can make sure a StageEntry is for the Proxy that's been renamed.  This is why the previous code ignores the old name and does not send it in the signal. I don't think you should send it down the signal and use it to identify the Stage Entry later, it's error-prone

Reply:
Yes of course I tested it and its working. The stage is not enough as a unique identifier, that's the whole point of this change. The previous code could ignore it since it doesn't matter. Since data which is stored in the combo box technically doesn't need updating (was just the stage), but in this new case it does, since the data in the combo box is the stage entry which also contains the name and the path.

Maybe an example could help. Lets assume you have the following proxyshapes (which are connected via the in stage data):
ProxyShape1 - Stage1
ProxyShape2 - Stage1

So lets assume to Rename "ProxyShape2" to "NewProxyShape2".

When you get to the part of the code which needs to update you will get two things:
oldName: ProxyShape2
newEntry: ("NewProxyShape2", "NewProxyShape2Path", Stage1)

So at this point I have to figure out which entry was changed and set the new data for it, since that data contains the name as well. And to find which entry in the list I need to know what the name use to be since if I just had the new Entry I would not be able to find it in the list. 

So since you are telling me that the displayname is not enough the other way to do this, but I'm not a fan of, is to use the proxyshapepath and ASSUME that the index of allStages maps to the dropBox list and just find the index of the NEW entry in that new list and just use that to update. I can make this change

Reply:
So I got rid of the old name downstream, but couldn't get rid of it here. Let me know if you have any ideas.

Reply:
To fix a different issue I was about to add another utility function in lib/mayaUsd/ufe/Utils.h to return the MDagPath of any given stage, that Utils class has access to the UsdStageMap we maintain.  A utility function like that might replace a lot of this wouldn't it?  The UsdStageMap should be able to correctly identify the dag object you need, but while testing my other issue I hadn't been thinking about multiple proxy shapes pointing to the same stage.  If the stage map doesn't solve your problem here then it also might not solve what I'm looking at in this situation.

Reply:
Since I'm not certain of MdagPath rules and unicity, they way I handled this in Bifrost is I would use the node's Muuid and store that as a string identifier.  you get it with MFnDependencyNode::uuid and then can call asString on the MUIid.  So the StageEntry could have also have an ProxyShape ID field that's the MUuid, or its string representation, and that's a reliable way to id that two objects are the same.  So you could pass down that ID in your rename signal instead of the previous name

Reply:
A second suggestion:  in this method, MayaSessionState::nodeRenamedCBOnIdle, you have the MObject, and if indeed you get MDagPath from it, you can then call MayaSessionState::getStageEntry to get a StageEntry for it which will run the same logic to generate the displayName, and then it should be much safer to compare the displayName

Reply:
I updated to use the UUID as you suggested, feels much more reliable now, great suggestion!